modules/plugins/utility/ccc/ccc.nix

```diff
@@ -1,10 +1,154 @@
 {lib, ...}: let
-  inherit (lib.options) mkEnableOption;
+  inherit (lib.options) mkOption mkEnableOption;
+  inherit (lib.types) listOf enum;
   inherit (lib.nvim.binds) mkMappingOption;
+  inherit (lib.nvim.types) mkPluginSetupOption luaInline;
+  inherit (lib.generators) mkLuaInline;
 in {
   options.vim.utility.ccc = {
     enable = mkEnableOption "ccc color picker for neovim";
 
+    setupOpts = mkPluginSetupOption "ccc.nvim" {
+      highlighter = mkOption {
+        type = luaInline;
+        default = mkLuaInline ''
+          {
+            auto_enable = true,
+            max_byte = 2 * 1024 * 1024, -- 2mb
+            lsp = true,
+            filetypes = colorPickerFts,
+          }
+        '';
+        description = "Settings for the highlighter";
+      };
+
+      pickers = mkOption {
+        type = luaInline;
+        default = mkLuaInline ''
+          {
+            ccc.picker.hex,
+            ccc.picker.css_rgb,
+            ccc.picker.css_hsl,
+            ccc.picker.ansi_escape {
+              meaning1 = "bright", -- whether the 1 means bright or yellow
+            },
+          }
+        '';
+        description = ''
+          List of formats that can be detected by |:CccPick| to be activated.
+        '';
+      };
+
+      alpha_show = mkOption {
+        type = enum [
+          "show"
+          "hide"
+          "auto"
+        ];
+        default = "hide";
+        description = ''
+          This option determines whether the alpha slider is displayed when the
+          UI is opened. "show" and "hide" mean as they are. "auto" makes the
+          slider appear only when the alpha value can be picked up.
+        '';
+      };
+
+      recognize = mkOption {
+        type = luaInline;
+        default = mkLuaInline ''
+          { output = true }
+        '';
```

Comment:
This could be either a list (so it converts to a table) or `luaInline`. I find `luaInline` to be generally confusing, so it *might* be nicer to go ahead and support lists *anyway* but it's not a blocker.